core/src/main/java/org/geysermc/geyser/registry/populator/CustomBlockRegistryPopulator.java

```diff
@@ -139,10 +132,27 @@ public void registerOverride(@NonNull JavaBlockState javaBlockState, @NonNull Cu
      * Registers all vanilla custom blocks and skulls defined by API and mappings
      */
     private static void populateVanilla() {
+        Int2ObjectMap<CustomBlockState> blockStateOverrides = new Int2ObjectOpenHashMap<>();
+
         for (CustomSkull customSkull : BlockRegistries.CUSTOM_SKULLS.get().values()) {
             CUSTOM_BLOCKS.add(customSkull.getCustomBlockData());
         }
 
+        for(Map.Entry<String, CustomBlockState> entry : BLOCK_STATE_OVERRIDES_QUEUE.entrySet()) {
```

Comment:
This queue should be cleared once we are done populating the vanilla registries.

Reply:
https://github.com/GeyserMC/Geyser/pull/4178/commits/f02a7776dae3013cc9d23a09e4689ba5cc56f152